### PR TITLE
feat(airflow): Add `--conf-source` to DAG created with `kedro-airflow`

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description

Resolve https://github.com/kedro-org/kedro-plugins/issues/672

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by ankatiyar** [comment]
Ready!

File: kedro-airflow/tests/test_plugin.py [hunk 1]
```diff
@@ -376,3 +376,19 @@ def test_create_airflow_all_and_pipeline(cli_runner, metadata):
...
+    result = cli_runner.invoke(commands, command, obj=metadata)
+    assert result.exit_code == 0
+    dag_file = metadata.project_path / "airflow_dags" / "fake_project_dag.py"
+
+    assert dag_file.exists()
+
+    expected_airflow_dag = f'conf_source = "{Path(metadata.project_path / "conf").resolve()}" or Path.cwd() / "conf"'
+    with dag_file.open(encoding="utf-8") as f:
+        dag_code = [line.strip() for line in f.read().splitlines()]
+        print(dag_code)
```
Comment (merelcht) [inline kedro-airflow/tests/test_plugin.py:392]: left-over print

**Comment by DimedS** [comment]
Great work, @ankatiyar ! The PR looks good, but I have one concern, I was following [deployment manual for cloud services](https://docs.kedro.org/en/stable/deployment/airflow.html) and tried to run 
`kedro airflow create --conf-source="plugins/conf-new_kedro_project.tar.gz"`
I received an error:
`Error: Invalid value for '--conf-source': Path 'plugins/conf-new_kedro_project.tar.gz' does not exist.`
I believe that's because I don't have that file in my current folder, but it will be on cloud container after it will be unpacked in plugins folder, so I think maybe we don't need that strict check on DAG creation step? 

Also it will be great to update deployment docs according to that new feature.

**Comment by ankatiyar** [comment]
@DimedS you're right, I turned of the check to see if the path exists.

**Comment by ankatiyar** [comment]
@DimedS Just wanted to check why resolved paths wouldn't work?

**Comment by DimedS** [comment]
> @DimedS Just wanted to check why resolved paths wouldn't work?

I think typically I'm creating a DAG in my local computer and it will be one absolute path and when I will run that DAG in some container in the cloud environment, absolute path to the same config will be different. So it's ok to use relative path, but not absolute.

**Review by DimedS — APPROVED** [review]
Great job, @ankatiyar !

**Pull request merged**